packages/backend/src/services/ServiceRepository.ts

```diff
@@ -1 +1,204 @@
+import type { GdriveService } from './GdriveService/GdriveService';
+import type { GithubAppService } from './GithubAppService/GithubAppService';
+import type { GitIntegrationService } from './GitIntegrationService/GitIntegrationService';
+import type { GroupsService } from './GroupService';
+import type { HealthService } from './HealthService/HealthService';
+import type { NotificationsService } from './NotificationsService/NotificationsService';
+import type { OrganizationService } from './OrganizationService/OrganizationService';
+import type { PersonalAccessTokenService } from './PersonalAccessTokenService';
+import type { PinningService } from './PinningService/PinningService';
+import type { ProjectService } from './ProjectService/ProjectService';
+import type { SavedChartService } from './SavedChartsService/SavedChartService';
+import type { SchedulerService } from './SchedulerService/SchedulerService';
+import type { SearchService } from './SearchService/SearchService';
+import type { ShareService } from './ShareService/ShareService';
+import type { SpaceService } from './SpaceService/SpaceService';
+import type { SshKeyPairService } from './SshKeyPairService';
+import type { UnfurlService } from './UnfurlService/UnfurlService';
+import type { UserAttributesService } from './UserAttributesService/UserAttributesService';
+import type { UserService } from './UserService';
+import type { ValidationService } from './ValidationService/ValidationService';
+
+/**
+ * Interface outlining all services available under the `ServiceRepository`. Add new services to
+ * this list (in alphabetical order, please!) to have typescript help ensure you've updated the
+ * service repository correctly.
+ */
+interface ServiceManifest {
+    analyticsService: AnalyticsService;
+    commentService: CommentService;
+    csvService: CsvService;
+    dashboardService: DashboardService;
+    downloadFileService: DownloadFileService;
+    encryptionService: EncryptionService;
+    gitIntegrationService: GitIntegrationService;
+    githubAppService: GithubAppService;
+    gdriveService: GdriveService;
+    groupService: GroupsService;
+    healthService: HealthService;
+    notificationService: NotificationsService;
+    organizationService: OrganizationService;
+    personalAccessTokenService: PersonalAccessTokenService;
+    pinningService: PinningService;
+    projectService: ProjectService;
+    savedChartService: SavedChartService;
+    schedulerService: SchedulerService;
+    searchService: SearchService;
+    shareService: ShareService;
+    sshKeyPairService: SshKeyPairService;
+    spaceService: SpaceService;
+    unfurlService: UnfurlService;
+    userAttributesService: UserAttributesService;
+    userService: UserService;
+    validationService: ValidationService;
+}
+
+/**
+ * Enforces the presence of getter methods for all services declared in the manifest.
+ */
+type ServiceFactoryMethod<T extends ServiceManifest> = {
```

Comment:
This is used to ensure you get a type error if `ServiceRepository` is not up to date with the service manifest above, and also to enforce the specific format of the method name (`get<serviceName` camelcased).

<img width="1119" alt="Screenshot 2024-03-04 at 20 37 40" src="https://github.com/lightdash/lightdash/assets/382538/2e8db406-b45f-4cb7-9dc5-e6c98ef1aad3">

Reply:
Nice typing 🏅